server/methods/messageSearch.js

```diff
@@ -226,7 +226,7 @@ Meteor.methods({
 				};
 			}
 
-			result.messages = RocketChat.models.Messages.find(query, options).fetch();
+			result.message.docs = RocketChat.models.Messages.find(query, options).fetch();
```

Comment:
any reason to change the result schema? I'm afraid this could break some clients. if there is no strong reason for this change, I would ask to stick if the old schema.

Reply:
The result schema has been changed because we wanted to allow some metadata to the search results (e.g. number of matching documents, scoring, etc) In order to keep one schema (necessary e.g. for result validation) for all providers, we changed it also here. Nevertheless, I changed this in the code in api, which uses his function, too . If you want to change there we can have the solutions: add a parameter *format* to the function, that is default *basic* but can be set to *extended*
What do you think?

Reply:
@tkurz what about adding another property to the result which is exclusively dedicated to the metadata? 
```javascript
result: {
  meta: {
    messages: {
      totalCount: 17177882,
      pagesize: 50,
      currentPage: 37
    },
    user: {
      language: 'prBR'
    }
}
```
I don't think it's a good style to change the schema of a response, but having optional parts for which a projection might be specified by the consumer is fine.

Reply:
Yes, would be a valid solution, but enforces some major changes in the current Chatpal implementation. What about this: Keep the "old" schema in the messageSearch Service and map it in the search provider. so we do not change the current api. IMHO the new schema is more adequate (and common) on search index systems  like solr/elastic. Would be easy and would not contain api changes (and this be downwards compatible to 3rd party)

Reply:
As you said, the new Schema us common for search providers. Since it’s the implementation of the „search“ service, I believe it’s valid to go for .docs - as you also changed the old global search implementation and the api to match, I don’t see a constraint to use this new schema. It was a beta in the current version explicitly anyway. But I think you should discuss this with @diegosampaio (probably on open.rocket.chat) during the course of today. 🇧🇷 ist close to waking up now anyway ;-)

Reply:
I'm here 👋  feel free to reach me out at https://open.rocket.chat/direct/diego.sampaio 

on the schema change, I just need to make sure it will at least not break our mobile apps and add a break changes alert on release notes..

Reply:
Hi @sampaiodiego ;) Regarding mobile apps: I can only find 2 usages of the messageSearch function:
1. DefaultSearchProvider (which can deal of course with the schema)
2. API: RocketChat.API 'chat.search' method (https://github.com/RocketChat/Rocket.Chat/blob/2d123a8be7ac8d4732214b13c8044fb70d382aac/packages/rocketchat-api/server/v1/chat.js#L123)

The PR already adapted this function to the new model:

https://github.com/redlink-gmbh/Rocket.Chat/blob/dfecb5e9ac1db98a8d74635153e9460321873528/packages/rocketchat-api/server/v1/chat.js#L140

Reply:
@rafaelks do the mobile apps utilize the real time method or the endpoint for searching? Or do they not search messages yet?

Reply:
@graywolf336 We're implementing using this API: https://rocket.chat/docs/developer-guides/rest-api/chat/search/.

Reply:
Thanks, I will take a look!

Reply:
@graywolf336 the rest endpoint uses still current search service (mongo based). And it returns the same result format as before. So schema change should not influence the app right? In a next version we can change the method (and maybe the api) so it supports both, basic and extended schema version. Then it would also be possible to use the current provider for API as well. But for now it should be fine right?

Reply:
yup.. seems we're good with mobile apps.. I'll still add a `[BREAK]` flag for third parties who still rely on DDP method.

looking forward to the next version with metadata on REST endpoints 😃